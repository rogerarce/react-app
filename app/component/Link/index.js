import React from 'react'

const STATUS = {
  NORMAL: 'normal',
  HOVERED: 'hovered',
}

export default class Link extends React.Component {
  constructor(props) {
    super(props)

    this._onMouseEnter = this._onMouseEnter.bind(this)
    this._onMouseLeave = this._onMouseLeave.bind(this)

    this.state = {
      class: STATUS.NORMAL
    }
  }
  
  _onMouseEnter() {
    this.setState({ class: STATUS.HOVERED })
  }
  
  _onMouseLeave() {
    this.setState({ class: STATUS.NORMAL })
  }

  render() {
    let { uri, children } = this.props

    return <a 
      className={this.state.class}
      onMouseEnter={this._onMouseEnter}
      onMouseLeave={this._onMouseLeave}
      href={uri}
    >{children}</a>
  }
}
