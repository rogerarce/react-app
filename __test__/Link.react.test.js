/*
 *
 */
import React from 'react'
import Link from '../app/component/Link/'
import renderer from 'react-test-renderer'

test('Link component sample test', () => {
  const component = renderer.create(
    <Link uri="https://facebook.com">Facebook</Link>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  tree.props.onMouseEnter();

  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  tree.props.onMouseLeave();

  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})
