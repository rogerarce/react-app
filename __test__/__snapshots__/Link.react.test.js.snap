// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link component sample test 1`] = `
<a
  className="normal"
  href="https://facebook.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Facebook
</a>
`;

exports[`Link component sample test 2`] = `
<a
  className="hovered"
  href="https://facebook.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Facebook
</a>
`;

exports[`Link component sample test 3`] = `
<a
  className="normal"
  href="https://facebook.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Facebook
</a>
`;
